cmd/frontend/graphqlbackend/schema.graphql

```
@@ -3279,6 +3314,11 @@ type ExternalService implements Node {
     """
     displayName: String!
 
+    """
+    The current rate limiter state for the external service.
+    """
+    rateLimiterState: RateLimiterState
```

maybe add a small comment what the `null` value indicates - or should this be non-nullable instead?

---

actually, reading the resolver code, I think we should return a ratelimiterstate for the default rate limiter we use instead, if possible. If nothing is configured, that does not mean infinite, it means default limit instead.

---

So in the UI, if it's `nil`, it simply won't show rate limiter status at all. It won't display infinite.

I looked at returning a default limiter state, but it gets a bit hairy with what would have to be returned. I also don't know if it would be confusing from a UI perspective if redis has a hiccup when you refresh, and for a brief moment your rate limit number is something completely different instead of simply not displaying 🤔

---

if redis is down, we should return an error instead, no?